ipatests/pytest_ipa/integration/create_bridge.py

```diff
@@ -0,0 +1,173 @@
+    host.run_command(["python", "manage.py", "migrate"],
+                     cwd=f"{dir}/src/ipa-tuura")
+
+    # Add necessary admin vars to bashrc
+    env_vars = textwrap.dedent(f"""
+    export DJANGO_SUPERUSER_PASSWORD={password}
+    export DJANGO_SUPERUSER_USERNAME=scim
+    export DJANGO_SUPERUSER_EMAIL=scim@ipa.test
+    """)
+
+    tasks.backup_file(host, '/etc/bashrc')
+    content = host.get_file_contents('/etc/bashrc', encoding='utf-8')
+    new_content = content + f"\n{env_vars}"
+    host.put_file_contents('/etc/bashrc', new_content)
+    host.run_command(['bash'])
+
+    # Create django admin
+    host.run_command(["python", "manage.py", "createsuperuser",
+                      "--scim_username", "scim", "--noinput"],
+                     cwd=f"{dir}/src/ipa-tuura")
+
+    # Open allowed hosts to any for testing
+    regex = r"^(ALLOWED_HOSTS) .*$"
+    replace = r"\1 = ['*']"
+    settings_file = f"{dir}/src/ipa-tuura/root/settings.py"
+    settings = host.get_file_contents(settings_file, encoding='utf-8')
+    new_settings = re.sub(regex, replace, settings, flags=re.MULTILINE)
+    host.put_file_contents(settings_file, new_settings)
+
+    # Setup keycloak service and config files
+    contents = textwrap.dedent(f"""
+    DJANGO_SUPERUSER_USERNAME=scim
+    DJANGO_SUPERUSER_PASSWORD={password}
+    DJANGO_SUPERUSER_EMAIL=scim@{host.domain.name}
+    """)
+    host.put_file_contents("/etc/sysconfig/scim", contents)
+
+    manage = f"{dir}/src/ipa-tuura/manage.py"
+    contents = textwrap.dedent(f"""
+    [Unit]
+    Description=SCIMv2 Bridge Server
+    After=network.target
+
+    [Service]
+    Type=idle
+    WorkingDirectory={dir}/src/ipa-tuura/
+    EnvironmentFile=/etc/sysconfig/scim
+    # Fix this later
+    # User=scim
+    # Group=scim
+    ExecStart=/usr/bin/python {manage} runserver 0.0.0.0:8000
+    TimeoutStartSec=600
+    TimeoutStopSec=600
+
+    [Install]
+    WantedBy=multi-user.target
+    """)
+    host.put_file_contents("/etc/systemd/system/scim.service", contents)
+    host.run_command(["systemctl", "daemon-reload"])
+    host.run_command(["systemctl", "start", "scim"])
+
+
+def setup_keycloak_scim_plugin(host, bridge_server):
+    dir = "/opt/keycloak"
+    password = host.config.admin_password
+
+    # Install needed packages
+    tasks.install_packages(host, ["unzip", "java-11-openjdk-headless",
+                                  "openssl", "maven"])
+
+    # Add necessary admin vars to bashrc
+    env_vars = textwrap.dedent(f"""
+    export KEYCLOAK_PATH={dir}
+    """)
+
+    tasks.backup_file(host, '/etc/bashrc')
+    content = host.get_file_contents('/etc/bashrc', encoding='utf-8')
+    new_content = content + f"\n{env_vars}"
+    host.put_file_contents('/etc/bashrc', new_content)
+    host.run_command(['bash'])
+
+    # Download keycloak plugin
+    zipfile = "scim-keycloak-user-storage-spi/archive/main.zip"
+    url = f"https://github.com/justin-stephenson/{zipfile}"
```

Comment:
Is it ok that this is unversioned? How will we know if/when it breaks whether main.zip was updated?

Reply:
I don't think there are any versioned branches/tags on that repo at this time.  I'll check if that can be added.

Reply:
Updated to use version tag